pkg/controller/middleware/debug.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/buildinfo"
+	"github.com/gorilla/mux"
+)
+
+// ProcessDebug adds additional debugging information to the response if the
+// request included the "X-Debug" header with any value.
+func ProcessDebug(ctx context.Context) mux.MiddlewareFunc {
+	header := "x-debug"
```

Comment:
nit: seems like this should be a const

Reply:
also - I'm not sure x-debug is the right header, maybe something like x-build-info ?

Reply:
Made them constants. The return headers are `x-buildinfo-*`. This is the _request_ header, and I'd like to give us flex to add other debug information if we wanted to in the future.